### ISSUE TITLE
Don't audit a shared object if it doesn't look like a Python extension

### ISSUE THREAD

**Issue by woodruffw** [trailofbits/abi3audit#12]
Not every shared object in a wheel is a Python extension. For example, a wheel might vendor a copy of a dependency so that the target host does not have to supply it. Tools like `auditwheel` automate this dependency vendoring.

As such, `abi3audit` should avoid false positives by limiting its search to only those shared objects that are *actually* native Python extensions. These can be identified by the following rules:

1. They have a filename like `foo(.abi3)?.{so,pyd}`
2. They have an *initialization function* with a public symbol of `PyInit_foo` or `PyInitU_foo`, where the latter is the PEP 489 format for unicode module names (`foo` is punycoded with `-` replaced with `_`)

In other words: if a shared object does not satisfy these conditions, then it is not a Python extension and should be skipped by `abi3audit`. In addition to eliminating potential false positives, this will speed up auditing a bit by excluding unnecessary files.

**Comment by woodruffw** [comment]
On further thought, there's too much nuance here to do this reliably: a wheel might have one extension module (`foo.abi3.so`) which then dynamically loads in other shared objects that have been vendored into the wheel, which would be entitled to use the CPython APIs.
